From 4f7f2cc750d1273782309d147c90ab8e4977bd23 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Wed, 5 Aug 2015 14:15:54 +0000
Subject: [PATCH/RFC] gitweb: Don't pass --full-history to git-log(1)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

When you look at the history for a file via "git log" we don't show
--full-history by default, but the Gitweb UI does so, which can be very
confusing for all the reasons discussed in "History Simplification" in
git-log(1) and in
http://thread.gmane.org/gmane.comp.version-control.git/89400/focus=90659

We've been doing history via --full-history since pretty much forever,
but I think this is much more usable, and on a typical project with lots
of branches being merged it makes for a much less confusing view. We do
this for git log by default, why wouldn't Gitweb follow suit?

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 gitweb/gitweb.perl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gitweb/gitweb.perl b/gitweb/gitweb.perl
index 7a5b23a..2913896 100755
--- a/gitweb/gitweb.perl
+++ b/gitweb/gitweb.perl
@@ -7387,7 +7387,7 @@ sub git_log_generic {
 	}
 	my @commitlist =
 		parse_commits($commit_hash, 101, (100 * $page),
-		              defined $file_name ? ($file_name, "--full-history") : ());
+		              defined $file_name ? $file_name : ());
 
 	my $ftype;
 	if (!defined $file_hash && defined $file_name) {
-- 
2.1.3

